### PR TITLE
Add Configurable Company API to wiki

### PR DESCRIPTION
**Pull request by TheAnsuz**
Adding [Configurable company API by Ansuz/AMRV](https://thunderstore.io/c/lethal-company/p/AMRV/ConfigurableCompany/) to the wiki since many developers may find it easier to understand it from here than from [thunderstore wiki](https://thunderstore.io/c/lethal-company/p/AMRV/ConfigurableCompany/wiki/) or [github wiki](https://github.com/TheAnsuz/Lethal-Company-Configurable-Company-API/wiki).

The wiki is part of the `Modding APIs` wiki section.

I reworked the contents of those existing wikis to provide more information in a friendly manner.

The API itself is not related to any other API in the wiki so I created both an overview and pages for each type of builder and complex feature that the developer might need to use the plugin correctly.

[Github](https://github.com/TheAnsuz/Lethal-Company-Configurable-Company-API)
[Thunderstore](https://thunderstore.io/c/lethal-company/p/AMRV/ConfigurableCompany/)

### CODE REVIEW

File: docs/dev/apis/configurable-company.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
+---
+prev: false
+next: false
```
Comment (Xilophor) [inline docs/dev/apis/configurable-company.md:3]: set to true for easy linking to the next page (developing pages)

File: docs/.vitepress/config.mts [hunk 2]
```diff
@@ -124,6 +124,29 @@ export default defineConfig({
                         { text: 'Custom Scrap', link: '/dev/apis/lethallib/customscrap'},
                       ]
                     },
+                    {
+                      text: 'Configurable Company',
+                      link: '/dev/apis/configurable-company',
+                      collapsed: true,
+                      items: [
+                        {
+                          text: 'Developing pages',
```
Comment (Xilophor) [inline docs/.vitepress/config.mts:133]: use Title Case for the article titles in the sidebar

File: docs/dev/apis/configurable-company/developing-pages.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
+---
+prev: false
```
Comment (Xilophor) [inline docs/dev/apis/configurable-company/developing-pages.md:2]: set to true to easily go back to the API's overview page

File: docs/dev/apis/configurable-company/developing-categories.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
+---
+prev: true
+next: true
+description: How to create categories for your plugin and how to use them.
```
Comment (Xilophor) [inline docs/dev/apis/configurable-company/developing-categories.md:4]: add 'Configurable Company' to this
...create Configurable Company categories...

File: docs/dev/apis/configurable-company/developing-configs.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
+---
+prev: true
+next: true
+description: How to create configurations for your plugin and how to use them.
```
Comment (Xilophor) [inline docs/dev/apis/configurable-company/developing-configs.md:4]: add 'Configurable Company' to this
...create Configurable Company configurations...

File: docs/dev/apis/configurable-company/developing-pages.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+next: true
+description: How to create pages for your plugin and how to use them.
```
Comment (Xilophor) [inline docs/dev/apis/configurable-company/developing-pages.md:4]: add 'Configurable Company' to this
...create Configurable Company pages...

File: docs/dev/apis/configurable-company/developing-categories.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+## Parameters
+
+-   `SetID(string)`: The unique ID of the category.
+-   `SetName(string)`: The name that will be displayed on the in-game menu.
+-   `SetColor(Color[white])`: (There are multiples methods to set the color. choose the prefered one based on your preferences) Sets the color of the category panel in the in-game menu.
+-   `HideIfEmpty(bool[true])`: Marks the category to not be displayed at all if there are no configurations inside this category.
+-   `SetPage(ConfigurationPage)`: Makes this category go listed on the provided page. One is generated by default but I encourage you to create a page for your mod.
+
+`Build()` Will create the category. It will not be created until the method is called (However as mentioned above, the method will called automatically if you assign the creation to a category).
```
Comment (Xilophor) [inline docs/dev/apis/configurable-company/developing-categories.md:48]: change Will to will

**Review by Xilophor — CHANGES_REQUESTED** [review]
Just needs some minor formatting & grammar changes

**Comment by TheAnsuz** [comment]
Updated :)

File: docs/dev/apis/configurable-company.md [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+```
+
+:::
+
+```cs
+[BepInPlugin(PluginInfo.PLUGIN_GUID, PluginInfo.PLUGIN_NAME, PluginInfo.PLUGIN_VERSION)]
+[BepInDependency( // [!code ++]
+    LethalConfiguration.PLUGIN_GUID, // [!code ++]
+    BepInDependency.DependencyFlags.HardDependency)] // [!code ++]
+    public class YourPlugin : BaseUnityPlugin {
```
Comment (Owen3H) [inline docs/dev/apis/configurable-company.md:87]: Missing closing bracket

File: docs/dev/apis/configurable-company.md [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+    LethalConfiguration.PLUGIN_VERSION
+```
+
+:::
+
+```cs
+[BepInPlugin(PluginInfo.PLUGIN_GUID, PluginInfo.PLUGIN_NAME, PluginInfo.PLUGIN_VERSION)]
+[BepInDependency( // [!code ++]
+    LethalConfiguration.PLUGIN_GUID, // [!code ++]
+    BepInDependency.DependencyFlags.HardDependency)] // [!code ++]
```
Comment (Owen3H) [inline docs/dev/apis/configurable-company.md:86]: Spanning this across multiple lines probably isn't the best bet for readability.

**Review by Xilophor — APPROVED** [review]
Lgtm

**Pull request merged**